tests/sanity_rgw.py

```diff
@@ -33,28 +40,98 @@ def run(ceph_cluster, **kw):
     rgw_node.exec_command(cmd='cd ' + test_folder + ' ; ' + git_clone)
     if ceph_cluster.containerized:
         test_folder_path = '/{test_folder}'.format(test_folder=test_folder)
-        rgw_ceph_object.exec_command(cmd='sudo rm -rf ' + test_folder)
-        rgw_ceph_object.exec_command(cmd='sudo mkdir ' + test_folder)
-        rgw_node.exec_command(
-            cmd='sudo docker cp {test_folder}/* {container}:/{test_folder}/'.format(
-                container=rgw_ceph_object.container_name,
-                test_folder=test_folder))
-        rgw_node.exec_command(cmd='curl https://bootstrap.pypa.io/get-pip.py -o ~/get-pip.py')
-        rgw_node.exec_command(
-            cmd='sudo docker cp ~/get-pip.py {container}:/get-pip.py'.format(container=rgw_ceph_object.container_name))
-        rgw_ceph_object.exec_command('python3 /get-pip.py')
-    rgw_ceph_object.exec_command(
-        cmd='sudo pip3 install -r ' + test_folder + '/ceph-qe-scripts/rgw/requirements.txt')
-    script_name = config.get('script-name')
-    config_file_name = config.get('config-file-name')
-    test_version = config.get('test-version', 'v2')
-    script_dir = DIR[test_version]['script']
-    config_dir = DIR[test_version]['config']
-    timeout = config.get('timeout', 300)
-    out, err = rgw_ceph_object.exec_command(
-        cmd='sudo python3 ' + test_folder_path + script_dir + script_name + ' -c '
-            + test_folder + config_dir + config_file_name,
-        timeout=timeout)
-    log.info(out.read().decode())
-    log.error(err.read().decode())
-    return 0
+        # rgw_ceph_object.exec_command(cmd='sudo rm -rf ' + test_folder)
+        # rgw_ceph_object.exec_command(cmd='sudo mkdir ' + test_folder)
+        test_cmd = 'python3 ' + test_folder_path + script_dir + script_name + ' -c ' \
+                   + test_folder + config_dir + config_file_name
+        build_a = ceph_cluster.rhcs_version
+        build = str(build_a)
+        distro_info = rgw_node.distro_info
+        distro_ver = distro_info['VERSION_ID']
+        if build.startswith('4'):
+            if distro_ver.startswith('8'):
```

Comment:
There is a lot of repetitive code here. please check again. there is definitely room for improvement

Reply:
@udaysk23 any update on this?

Reply:
I still fee there is a room for reducing all these `if`s. but for now It's okay, maybe we can revist again later